app/src/main/java/fr/free/nrw/commons/contributions/RxJava2Tasks.java

```diff
@@ -0,0 +1,30 @@
+package fr.free.nrw.commons.contributions;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Locale;
+
+import fr.free.nrw.commons.PageTitle;
+import io.reactivex.Single;
+
+class RxJava2Tasks {
+
+    private static final String UPLOAD_COUNT_URL_TEMPLATE =
+            "https://tools.wmflabs.org/urbanecmbot/uploadsbyuser/uploadsbyuser.py?user=%s";
+
+    static Single<Integer> getUploadCount(String userName) {
+        return Single.fromCallable(() -> {
+            URL url = new URL(String.format(Locale.ENGLISH, UPLOAD_COUNT_URL_TEMPLATE,
+                    new PageTitle(userName).getText()));
+            HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();
+            BufferedReader bufferedReader = new BufferedReader(new
+                    InputStreamReader(urlConnection.getInputStream()));
+            String uploadCount = bufferedReader.readLine();
```

Comment:
No problem at all with the `leaking` concept :) My jokes are sometimes pretty bad (but they are still jokes in my heart).

In this kind of scenario I always think about what the client can do and how to serve it best.
The client expects one result and only one, so a Single is the clear choice. I could use an Observable for everything, but a Single will make the purpose of the task clear and avoid any unnecessary overhead and/or complexity. 
i think we agree on this 👍 .

Now, what about failures?
As a client, I don't care. It is not my responsibility. 
You (the Single<>) can either give me the one result I want, or you fail. I don't care about the cause, I don't want to know how you obtain the result.
So from this perspective, the Single (sorry for the bad wording) should try to fix whatever issue it finds. It it can't solve the issue, it is a failure and it should tell the client. Then the client can decide whether to give the Single another chance, use another resource, or give up.
The client should not care about why the Single failed, should not try to solve the issue for the Single.

I believe this is why the `onError` is in the `subscribe` and nowhere else. Because if an error can be handled upstream I don't care about it, solve it for me; if it can't be handled, just tell me you couldn't do your job and I'll decide who to give that task to now.

I see where the reasoning you follow, but I think there's no practical benefit on it. That is why I asked for some example code, because that will completely clarify whether there is any benefit on your proposal.
As of now, I can only see it adding more complexity...